parser.go

```diff
@@ -704,7 +704,12 @@ func Delete(data []byte, keys ...string) []byte {
 		newOffset = prevTok + 1
 	}
 
-	data = append(data[:newOffset], data[endOffset:]...)
+	// We have to make a copy here if we don't want to mangle the original data, because byte arrays are
```

Comment:
`byte slices are accessed by reference` it is a slice not an array which is pretty key distinction regarding reference/value passing.

Reply:
whoops, thanks for the catch. I'll fix